src/ucs/debug/instrument.h

```diff
@@ -97,11 +119,14 @@ void __ucs_instrument_record(uint64_t location, uint64_t lparam, uint32_t wparam
  */
 #define UCS_MAKE_LABEL(uniq) \
     label_##uniq
-#define UCS_INSTRUMENT_RECORD_(uniq, lparam, wparam, ...) \
+#define UCS_INSTRUMENT_RECORD_(uniq, type, name, lparam, wparam, ...) \
     do { \
-        if (ucs_instr_ctx.enable) do { \
+        if (ucs_instr_ctx.enabled[type]) do { \
     UCS_MAKE_LABEL(uniq):\
-            __ucs_instrument_record((uint64_t)&&UCS_MAKE_LABEL(uniq), (uint64_t)(lparam), (uint32_t)(wparam)); \
+            __ucs_instrument_record((uint64_t)&&UCS_MAKE_LABEL(uniq), \
+                    (uint64_t)(lparam), (uint32_t)(wparam)); \
+            __ucs_instrument_register((uint64_t)&&UCS_MAKE_LABEL(uniq), \
```

Comment:
also - you are adding array search for every instrumentation , this is a significant overhead